File: app/src/main/java/com/thewizrd/simpleweather/fragments/WindowColorFragment.java
package com.thewizrd.simpleweather.fragments;

import android.os.Bundle;

import androidx.annotation.CallSuper;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.DefaultLifecycleObserver;
import androidx.lifecycle.LifecycleOwner;

import com.thewizrd.simpleweather.helpers.WindowColorManager;

public abstract class WindowColorFragment extends CustomFragment implements WindowColorManager {

    @CallSuper
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getLifecycle().addObserver(new DefaultLifecycleObserver() {
            @Override
            public void onStart(@NonNull LifecycleOwner owner) {
                DefaultLifecycleObserver.super.onStart(owner);
                updateWindowColors();
            }
        });
    }

    public abstract void updateWindowColors();
}

File: app/src/main/java/com/thewizrd/simpleweather/fragments/WindowColorFragment.kt
package com.thewizrd.simpleweather.fragments

import android.os.Bundle
import androidx.annotation.CallSuper
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import com.thewizrd.simpleweather.helpers.WindowColorManager
import kotlinx.coroutines.launch

abstract class WindowColorFragment : CustomFragment(), WindowColorManager {
    @CallSuper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED) {
                updateWindowColors()
            }
        }
    }

    abstract override fun updateWindowColors()
}